eitprocessing/datahandling/mixins/equality.py

```diff
@@ -106,8 +106,22 @@ def isequivalent(self, other: Self, raise_: bool = False) -> bool:  # noqa: C901
 
             # check attributes of data
             else:
-                self._check_equivalence: list[str]
-                for attr in self._check_equivalence:
+                if is_dataclass(self):
+                    check_equivalence_fields = filter(
+                        lambda x: "check_equivalence" in x.metadata and x.metadata["check_equivalence"],
```

Comment:
I guess the "check_equivalence" metadata field will never exist if it's not set to True, so might not be necessary to check both it's existence and it's setting.
Then again, maybe it's safer to check for it just in case some edge case is created in the future where it might be flexible...

(I guess this is an odd comment, no action needed. I just had a secondary thought as I was typing it and figured I'd leave it here just to have it documented 😄)

Reply:
I think it's always best to do the explicit test. For instance when someone wants to see what happens if you change it to False; without checking the actual value this would not change behaviour, which is unexpected.